### ISSUE TITLE
implement a softmax version of focal loss

### ISSUE THREAD

**Issue by wyli** [Project-MONAI/MONAI#6510]
**Is your feature request related to a problem? Please describe.**
```
The softmax version of focal loss is:

  FL(p_t) = -alpha * (1 - p_t)**gamma * log(p_t),

where p_i = exp(s_i) / sum_j exp(s_j), t is the target (ground truth) class, and
s_j is the unnormalized score for class j.
```

cf https://github.com/pytorch/pytorch/blob/main/modules/detectron/softmax_focal_loss_op.cc#L38-L43

the current implementation is hard-coded to use sigmoid https://github.com/Project-MONAI/MONAI/blob/e5cf74aac60cb3512d92f9abb3dd48b5ed36e8d0/monai/losses/focal_loss.py#L26

**Comment by qingpeng9802** [comment]
I can work on this issue and PR later if no one is working on this.
